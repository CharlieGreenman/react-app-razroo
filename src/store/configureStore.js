import {createStore, applyMiddleware} from "redux";
import { createEpicMiddleware } from 'redux-observable';
import rootReducer from "../reducers/index";
import rootEpic from "../epics/index";

const epicMiddleware = createEpicMiddleware(rootEpic);
const createStoreWithMiddleware = applyMiddleware(epicMiddleware)(createStore);


export default function configureStore(initialState) {
    const store = createStoreWithMiddleware(rootReducer, initialState,
       window.devToolsExtension ? window.devToolsExtension() : undefined
    );
    return store;
}
