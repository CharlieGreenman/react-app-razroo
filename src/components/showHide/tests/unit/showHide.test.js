import React from 'react';
import ReactDOM from 'react-dom';
import Form from "../../showHide.js";

import {Provider} from "react-redux";
import configureStore from "../../../../store/configureStore";
const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <Form />
    </Provider>
    , div
  );
});
