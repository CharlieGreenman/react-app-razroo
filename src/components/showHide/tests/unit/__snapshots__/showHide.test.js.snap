// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="works"
  >
    <button
      className="show-hide-button"
      id="show-block"
      onClick={[Function]}
      type="button"
    >
      show
    </button>
    <button
      className="show-hide-button"
      id="hide-block"
      onClick={[Function]}
      type="button"
    >
      Hide
    </button>
  </div>
</div>
`;
